CMakeLists.txt

```diff
@@ -52,27 +52,135 @@ target_link_libraries(logging_client PRIVATE
   http_parser.host
 )
 
-add_enclave_lib(loggingenc
-  SRCS
-    src/apps/logging/logging.cpp
-    src/apps/logging/stub_for_code_signing.cpp
-)
-sign_app_library(loggingenc
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/apps/logging/oe_sign.conf
-  ${CMAKE_CURRENT_SOURCE_DIR}/src/apps/sample_key.pem
-)
+# Build common library for CCF enclaves
+add_custom_target(ccfcommon ALL)
+
+if("sgx" IN_LIST TARGET)
+  # enclave version
+  add_library(ccfcommon.enclave STATIC
+    ${CCF_DIR}/src/enclave/main.cpp
+    ${CCF_GENERATED_DIR}/ccf_t.cpp
+  )
+
+  target_compile_definitions(ccfcommon.enclave PUBLIC
+    INSIDE_ENCLAVE
+    _LIBCPP_HAS_THREAD_API_PTHREAD
+  )
+
+  target_compile_options(ccfcommon.enclave PUBLIC
+    -nostdinc
+    -nostdinc++
+  )
+
+  target_include_directories(ccfcommon.enclave SYSTEM PUBLIC
+    ${EVERCRYPT_INC}
+    ${CCF_GENERATED_DIR}
+  )
+
+  add_dependencies(ccfcommon.enclave flatbuffers)
+
+  if (PBFT)
+    target_link_libraries(ccfcommon.enclave PUBLIC
+      libbyz.enclave
+    )
+  endif()
+
+  target_link_libraries(ccfcommon.enclave PUBLIC
+    openenclave::oeenclave
+    openenclave::oelibcxx
+    ccfcrypto.enclave
+    evercrypt.enclave
+    http_parser.enclave
+    lua.enclave
+    secp256k1.enclave
```

Comment:
Should we have quickjs.enclave in here too?

Reply:
No. Currently most of the apps don't require `quickjs`, only `jsgeneric` does. They all require `lua` for the member governance. When we implement the member governance in JS, this should switch from a common `lua` dependency to a common `quickjs` dependency (or, perhaps we factor out the member governance frontend from the common framework which requires neither).